packages/asset/contracts/AssetReveal.sol

```diff
@@ -332,23 +328,21 @@ contract AssetReveal is IAssetReveal, Initializable, ERC2771Handler, EIP712Upgra
 
     /// @notice Checks if each metadatahash has been used before to either get the tokenId that was already created for it or generate a new one if it hasn't
     /// @dev This function also validates that we're not trying to reveal a tokenId that has already been revealed
-    /// @param amounts The amounts of tokens to mint
     /// @param metadataHashes The hashes of the metadata
     /// @param prevTokenId The previous token id from which the assets are revealed
     /// @return tokenIdArray The array of tokenIds to mint
-    function getRevealedTokenIds(
-        uint256[] calldata amounts,
-        string[] calldata metadataHashes,
-        uint256 prevTokenId
-    ) internal returns (uint256[] memory) {
+    function getRevealedTokenIds(string[] calldata metadataHashes, uint256 prevTokenId)
+        internal
+        returns (uint256[] memory)
+    {
         IAsset.AssetData memory data = prevTokenId.getData();
         require(!data.revealed, "Asset: already revealed");
-
-        uint256[] memory tokenIdArray = new uint256[](amounts.length);
-        for (uint256 i = 0; i < amounts.length; i++) {
+        uint256[] memory tokenIdArray = new uint256[](metadataHashes.length);
+        for (uint256 i = 0; i < metadataHashes.length; i++) {
             uint256 tokenId = assetContract.getTokenIdByMetadataHash(metadataHashes[i]);
             if (tokenId != 0) {
-                tokenId = assetContract.getTokenIdByMetadataHash(metadataHashes[i]);
+                tokenIdArray[i] = tokenId;
+                continue;
             } else {
                 uint16 revealNonce = ++revealIds[data.creator][prevTokenId];
```

Comment:
have we got tests for reveal nonce incrementation?

Reply:
Tests for revealing need more updates, ive noted this comment in a todo

Reply:
Ok!